// @visual-filter/common/dist/index.js
"use strict";
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.deepCopy = exports.DataType = exports.GroupType = exports.FilterType = void 0;
var FilterType;
(function (FilterType) {
    FilterType["GROUP"] = "group";
    FilterType["CONDITION"] = "condition";
})(FilterType = exports.FilterType || (exports.FilterType = {}));
var GroupType;
(function (GroupType) {
    GroupType["AND"] = "and";
    GroupType["NOT_AND"] = "not and";
    GroupType["OR"] = "or";
    GroupType["NOT_OR"] = "not or";
})(GroupType = exports.GroupType || (exports.GroupType = {}));
var DataType;
(function (DataType) {
    DataType["NUMERIC"] = "numeric";
    DataType["NOMINAL"] = "nominal";
})(DataType = exports.DataType || (exports.DataType = {}));
function deepCopy(src) {
    if (src.constructor === Object) {
        return Object.entries(src).reduce(function (objCopy, _a) {
            var _b;
            var key = _a[0], value = _a[1];
            return (__assign(__assign({}, objCopy), (_b = {}, _b[key] = deepCopy(value), _b)));
        }, {});
    }
    if (src.constructor === Array) {
        return src.map(deepCopy);
    }
    return src.valueOf();
}
exports.deepCopy = deepCopy;
